### ISSUE TITLE
Entity author

### ISSUE THREAD

**Issue by PekkaJalonen** [misje/opencti-wazuh-connector#48]
Connector replaces entity author when enrichment is applied. This is problematic example if author is used to fetch entities first, then apply enrichment, the author is replaced with connector author.

Case example: I want to query Opencti for CISA vulnerabilities, and run enrichment. Then, if later as wazuh collects more data, and I want to apply re-enrichment, the CISA list is incomplete as author has changed to wazuh connector. 

Would it be possible to add option to not change authors of existing data. We can still see labels which has been applied.

**Comment by misje** [comment]
Replacing the author is an issue I can definitely understand. I'm bringing up a test instance with the Mitre data set imported for testing. Unless reproducing this issue is as simple as enriching a vulnerability imported by Mitre, for a user with confidence set to 100, I appreciate information about the steps needed to reproduce.

**Comment by PekkaJalonen** [comment]
One thing that I forgot to mention, replacement happens when there is a match of vulnerability. It does not replace when there is no match.

**Comment by PekkaJalonen** [comment]
I just caught up with the breaking change https://blog.filigran.io/breaking-change-evolution-of-the-way-connector-streams-and-feeds-import-data-in-opencti-d10d7eb4407e of confidence level for connector user. So if mitre user confidence level is 80 and wazuh 100, it will overwrite. I did not test what happens if the level is 80 for both. There is also new variable default_score implemented for some connectors, which can ne used to set the connector confidence level.

**Comment by misje** [comment]
If I create a vulnerability that I know exists in Wazuh, I don't see the author being replaced. Even after a second enrichment. Only the CVSS details are updated. How do I reproduce this?

In the latest versions of OpenCTI, you may configure confidence down to per-entity level. You can give this connector, say 80 in confidence, but less for Vulnerability entities. That way, it will never update any fields on those entities. By the way, the connector doesn't change the vulnerability description, just CVSS fields.

The Figma diagram on [this page](https://docs.opencti.io/latest/usage/reliability-confidence/) in the documentation describes in detail how confidence affects updates to entities.

**Comment by misje** [comment]
I can't reproduce this. What version of OpenCTI are you using? OpenCTI fixes (and creates) a lot of bugs all the time. Perhaps this is an OpenCTI issue?

**Comment by PekkaJalonen** [comment]
Hi, I think that the reason was the connect user that I was using, it had confidence level 100 while mitre has 80. After I set the confidence level to same for mitre, now I cannot reproduce the issue. Also watch I see was that the Vulnerability TLP was change to what was set for the wazuh connectors  TLP.

I did run enrichment from opencti entities filtering vulnerabilities and selecting few, not by selecting single vulnerability and executing one-by-one, so handling it over the Processing\Tasks might have impacted this.
